src/Giraffe/Routing.fs

```diff
@@ -65,8 +71,9 @@ type HttpContext with
     member this.SetResolvedPath (path : string) =
         let subRoutingFeature = this.Features.Get<ISubRoutingFeature>()
         subRoutingFeature.SetResolvedPath path
-        // Keep this as a fallback for Saturn, etc?
-        this.Items.[RouteKey] <- path
+
+        if subRoutingFeature.CompatibilityMode = Version36 then
+            this.Items.[RouteKey] <- path
```

Comment:
Same here, only populating the `Items` key-value pair if running in `Version36` mode.